src/stdlib/base.rs

```diff
@@ -212,13 +212,33 @@ pub fn load_base<'gc>(ctx: Context<'gc>) {
                 Value::UserData(u) => u.metatable(),
                 _ => None,
             } {
+                /// Simply matches PUC-Rio behavior of returning the first 3 elements of the __pairs metacall
+                #[derive(Collect)]
+                #[collect(require_static)]
+                struct PairsReturn;
+
+                impl<'gc> Sequence<'gc> for PairsReturn {
+                    fn poll(
+                        &mut self,
+                        _ctx: Context<'gc>,
+                        _exec: Execution<'gc, '_>,
+                        mut stack: Stack<'gc, '_>,
+                    ) -> Result<SequencePoll<'gc>, Error<'gc>> {
+                        // QUESTION how do you just consume the data w/o doing anything?
+                        if stack.len() > 3 {
+                            _ = stack.drain(3..).count();
+                        }
```

Comment:
The `drain` iterator will remove the values when dropped, even if the iterator is not consumed, so you can just do `stack.drain(3..)`

Reply:
Awesome thanks!